# Bank Customer Churn Analysis

## Project Overview

This project aims to analyze customer data and identify factors
associated with customer churn.

## Objective

The objective of this notebook is to understand the structure,
content, and quality of the raw dataset before data cleaning
and exploratory data analysis.

In [4]:
import pandas as pd
pd.set_option("display.max_columns", None)

In [2]:
# Import libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 2. Load Data set

# what's sheet on this data.

In [3]:
excel_file = pd.ExcelFile(
    "C:/Users/Redmi/.vscode/Bank_churn/Bank_Churn_Messy.xlsx"
)

print(excel_file.sheet_names)

['Customer_Info', 'Account_Info']


# Check data for a specific sheet

In [4]:
# Read Customer_Info sheet
df_customer = pd.read_excel(
    "C:/Users/Redmi/.vscode/Bank_churn/Bank_Churn_Messy.xlsx",
    sheet_name="Customer_Info"
)

# Read Account_Info sheet
df_account = pd.read_excel(
    "C:/Users/Redmi/.vscode/Bank_churn/Bank_Churn_Messy.xlsx",
    sheet_name="Account_Info"
)

print("Customer_Info:")
display(df_customer.head())

print("Account_Info:")
display(df_account.head())

Customer_Info:


,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,EstimatedSalary
0,15634602,Hargrave,619,FRA,Female,42.0,2,€101348.88
1,15647311,Hill,608,Spain,Female,41.0,1,€112542.58
2,15619304,Onio,502,French,Female,42.0,8,€113931.57
3,15701354,Boni,699,FRA,Female,39.0,1,€93826.63
4,15737888,Mitchell,850,Spain,Female,43.0,2,€79084.1


Account_Info:


,CustomerId,Balance,NumOfProducts,HasCrCard,Tenure,IsActiveMember,Exited
0,15634602,€0.0,1,Yes,2,Yes,1
1,15634602,€0.0,1,Yes,2,Yes,1
2,15647311,€83807.86,1,Yes,1,Yes,0
3,15619304,€159660.8,3,No,8,No,1
4,15701354,€0.0,2,No,1,No,0


## 3. Dataset Overview

In [9]:
# Check the number of rows and columns on Customer_Info sheet

print("Number of rows:", df_customer.shape[0])
print("Number of columns:", df_customer.shape[1])

# Display column names of customer sheet

print("Column names:")
print(df_customer.columns.tolist())

# Display dataset information

print("Customer_Info sheet information")
df_customer.info()




# Check the number of rows and columns on Account_Info sheet 

print("Number of rows:", df_account.shape[0])
print("Number of columns:", df_account.shape[1])

# Display column names of account sheet

print("Column names:")
print(df_account.columns.tolist())

# Display dataset information

print("Account_Info sheet information")
df_account.info()

Number of rows: 10001
Number of columns: 8
Column names:
['CustomerId', 'Surname', 'CreditScore', 'Geography', 'Gender', 'Age', 'Tenure', 'EstimatedSalary']
Customer_Info sheet information
<class 'pandas.DataFrame'>
RangeIndex: 10001 entries, 0 to 10000
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   CustomerId       10001 non-null  int64  
 1   Surname          9998 non-null   str    
 2   CreditScore      10001 non-null  int64  
 3   Geography        10001 non-null  str    
 4   Gender           10001 non-null  str    
 5   Age              9998 non-null   float64
 6   Tenure           10001 non-null  int64  
 7   EstimatedSalary  10001 non-null  str    
dtypes: float64(1), int64(3), str(4)
memory usage: 625.2 KB
Number of rows: 10002
Number of columns: 7
Column names:
['CustomerId', 'Balance', 'NumOfProducts', 'HasCrCard', 'Tenure', 'IsActiveMember', 'Exited']
Account_Info sheet information
<class 'pand

based on data types:
the data type of the Age is float64 _ invalid and need to check
the data type of EstimateSalary column is str _ invalid and need to check

## 4. Data Quality Assessment

In [20]:
# Check missing values

missing_values = df.isnull().sum()

print("Missing values:")
print(missing_values)

Missing values:
CustomerId         0
Surname            3
CreditScore        0
Geography          0
Gender             0
Age                3
Tenure             0
EstimatedSalary    0
dtype: int64


In [16]:
# Check duplicated rows

duplicate_rows = df.duplicated().sum()

print("Number of duplicated rows:", duplicate_rows)

Number of duplicated rows: 1


In [17]:
# Check the number of unique values in each column

unique_values = df.nunique()

print("Number of unique values:")
print(unique_values)

Number of unique values:
CustomerId         10000
Surname             2932
CreditScore          460
Geography              5
Gender                 2
Age                   70
Tenure                11
EstimatedSalary     9997
dtype: int64


## 5. Summary Statistics

In [18]:
# Generate descriptive statistics for numerical variables

df.describe()

,CustomerId,CreditScore,Age,Tenure
count,1.000100e+04,10001.000000,9998.000000,10001.000000
mean,1.569093e+07,650.542946,38.920984,5.012699
std,7.193531e+04,96.658818,10.489116,2.892047
min,1.556570e+07,350.000000,18.000000,0.000000
25%,1.562852e+07,584.000000,32.000000,3.000000
50%,1.569073e+07,652.000000,37.000000,5.000000
75%,1.575323e+07,718.000000,44.000000,7.000000
max,1.581569e+07,850.000000,92.000000,10.000000


### Initial Observations

- The dataset contains approximately 10,000 customer records.
- `CustomerId` has 10,000 unique values and is an identifier variable.
- `Surname` has 2,932 unique values and may contain repeated customer surnames.
- `Geography` contains 5 unique categories.
- `Gender` contains 2 unique categories.
- `Age` ranges from 18 to 92 years.
- `Tenure` ranges from 0 to 10 years.
- The `Age` variable contains missing values because its count is lower than the total number of records.
- The `Tenure` variable has more observations than the expected dataset size, which should be investigated further.

In [7]:
print("Dataset shape:")
print(df.shape)

print("\nMissing values:")
print(df.isnull().sum())

print("\nDuplicated rows:")
print(df.duplicated().sum())

Dataset shape:
(10001, 8)

Missing values:
CustomerId         0
Surname            3
CreditScore        0
Geography          0
Gender             0
Age                3
Tenure             0
EstimatedSalary    0
dtype: int64

Duplicated rows:
1
